sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -617,91 +617,107 @@ object DateTimeUtils {
     (date - localDate.getDayOfMonth) + localDate.lengthOfMonth()
   }
 
-  // Visible for testing.
-  private[sql] val TRUNC_TO_YEAR = 1
-  private[sql] val TRUNC_TO_MONTH = 2
-  private[sql] val TRUNC_TO_QUARTER = 3
-  private[sql] val TRUNC_TO_WEEK = 4
-  private[sql] val TRUNC_TO_DAY = 5
-  private[sql] val TRUNC_TO_HOUR = 6
-  private[sql] val TRUNC_TO_MINUTE = 7
-  private[sql] val TRUNC_TO_SECOND = 8
+  // The constants are visible for testing purpose only.
   private[sql] val TRUNC_INVALID = -1
+  // The levels from TRUNC_TO_WEEK to TRUNC_TO_MILLENNIUM are used in truncations
+  // of DATE and TIMESTAMP values.
+  private[sql] val TRUNC_TO_WEEK = 0
+  private[sql] val TRUNC_TO_MONTH = 1
+  private[sql] val TRUNC_TO_QUARTER = 2
+  private[sql] val TRUNC_TO_YEAR = 3
+  private[sql] val TRUNC_TO_DECADE = 4
+  private[sql] val TRUNC_TO_CENTURY = 5
+  private[sql] val TRUNC_TO_MILLENNIUM = 6
+  private[sql] val MAX_LEVEL_OF_DATE_TRUNC = TRUNC_TO_MILLENNIUM
```

Comment:
I'm a little bit confused about the word `MAX` here. `MAX` indicates that these units are comparable, which means `TRUNC_TO_MONTH(1) > TRUNC_TO_WEEK(0)`, and `TRUNC_TO_DAY(12) > TRUNC_TO_MONTH(1)`. It's a little counter-intuitive because naturally `MONTH > WEEK` and `MONTH > DAY`.

Reply:
What would you propose? What about `LAST` like `LAST_LEVEL_OF_DATE_TRUNC`?

Reply:
We can order it from `TRUNC_TO_MILLENNIUM` to `TRUNC_TO_MICROSECOND`, and make `MIN_LEVEL_OF_DATE_TRUNC= TRUNC_TO_WEEK`?